client/js/components/App.react.jsx

```diff
@@ -79,13 +79,20 @@ export default class App extends React.Component {
         this.setState({allParks: parksWithDistance});
     }
 
-    applyFilters(filter) {
+    applyFilters(filters) {
         var visibleParks = this.state.allParks.filter((park) => {
             // FIXME: Convert park ids to numbers when we generate the data
-            var matchingAmenity = !!this.state.amenityLookup && this.state.amenityLookup[filter] && this.state.amenityLookup[filter].map(Number).indexOf(park.park_id) !== -1;
-            var matchingFacility = !!this.state.facilityLookup && this.state.facilityLookup[filter] && this.state.facilityLookup[filter].map(Number).indexOf(park.park_id) !== -1;
-
-            return matchingAmenity || matchingFacility;
+            return filters.reduce( (matches, filter) => {
+                const matchingFacility =
```

Comment:
In the reduction function, `matchingFacility` and `matchingAmenity` are defined once and have block scope due to `const`. This means their lifespan is just for that reduce iteration. I try to use const and let instead of var because javascript hoisting can cause confusion.